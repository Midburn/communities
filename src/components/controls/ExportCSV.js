import React from 'react';
import { CSVLink } from 'react-csv';
import { MDBIcon, MDBTooltip } from 'mdbreact';
import { withI18n } from 'react-i18next';

const CSV = (props) => {
    const {t, data, filename} = props;
    return (
        <div>
            <CSVLink data={data}
                     filename={filename}
                     target="_blank">
                <MDBTooltip
                    placement={`top`}
                    component="button"
                    componentClass="btn btn-primary"
                    tag="div"
                    color={'primary'}
                    tooltipContent={t('export')}>
                    <MDBIcon size="lg" icon="file-excel-o"></MDBIcon>
                </MDBTooltip>
            </CSVLink>
        </div>

    );
};
export const ExportCSV = withI18n()(CSV);