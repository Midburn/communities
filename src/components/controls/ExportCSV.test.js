import React from 'react';
import ReactDOM from 'react-dom';
import {ExportCSV} from './ExportCSV'
import { shallow } from 'enzyme';

describe('ExportCSV Component', () => {
    const TEST_TITLE = 'Test Title';
    const translateFunction = (term) => TEST_TITLE;
    it('should render correctly with sample properties', () => {
        const component = shallow(<ExportCSV t={translateFunction} data={{}} filename="Test"/>);
        expect(component).toMatchSnapshot();
    });
});