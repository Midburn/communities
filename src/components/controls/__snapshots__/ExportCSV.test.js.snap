// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportCSV Component should render correct translated title 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ExportCSV Component should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ExportCSV Component should render correctly with sample properties 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
